src/routes/accounts.py

```
@@ -1,25 +1,244 @@
+    )
+    if expires_at < datetime.now(timezone.utc):
+        raise HTTPException(
+            status_code=400, detail="Invalid or expired activation token."
+        )
+    user_status.is_active = True
+    await db.commit()
+    await db.refresh(user_status)
+    await db.delete(db_token)
+    await db.commit()
+
+    return {"message": "User account activated successfully."}
+
+
+@router.post("/password-reset/request/", response_model=dict)
+async def password_reset_token(
+    model: TokenPasswordRefresh, db: AsyncSession = Depends(get_db)
+):
+    user = await get_user_by_email(db, model.email)
+    if user and user.is_active:
+        old_tokens = await db.execute(
+            select(PasswordResetTokenModel).where(
+                PasswordResetTokenModel.user_id == user.id
+            )
+        )
+        for token in old_tokens.scalars().all():
+            await db.delete(token)
+        await db.commit()
+        new_refresh_token = PasswordResetTokenModel(user_id=user.id)
+        db.add(new_refresh_token)
+        await db.commit()
+
+    return {
+        "message": "If you are registered, you will receive an email with instructions."
+    }
+
+
+@router.post("/reset-password/complete/", response_model=dict)
+async def password_reset_compleat(
+    model: ResetPassword, db: AsyncSession = Depends(get_db)
+):
+    try:
+        user = await get_user_by_email(db, model.email)
+        check_token = await db.execute(
+            select(PasswordResetTokenModel).where(
+                PasswordResetTokenModel.token == model.token
+            )
+        )
+        if user is None:
+            raise HTTPException(
+                status_code=400, detail="Invalid email or token."
+            )
+        db_check_token = check_token.scalar_one_or_none()
+        if db_check_token is None:
+            await remove_token(db, user)
+            raise HTTPException(
+                status_code=400, detail="Invalid email or token."
+            )
```

In the password-reset completion flow you fetch the token by its value but never verify that the found token belongs to the `user` looked up by email. Add a check like `if db_check_token.user_id != user.id: await remove_token(db, user); raise HTTPException(status_code=400, detail="Invalid email or token.")`. Without this ownership check a token for a different user could be accepted. Also ensure you check `user is None` before proceeding (you already do - good).